event_management/src/main/java/com/example/event_management/service/UserEventService.java

```diff
@@ -15,35 +19,54 @@ public class UserEventService {
     @Autowired
     UserEventRepository userEventRepository;
 
-    public void deleteByEventId(String id) {
-        userEventRepository.deleteByEventId(id);
+    @Autowired
+    EventsRepository eventsRepository;
+
+    @Test
+    @Transactional
+    public void deleteByEventId(String eventId) {
+
+        userEventRepository.deleteByEventId(eventId);
     }
 
-    public void deleteByUserId(String id) {
-        userEventRepository.deleteByUserId(id);
+    @Test
+    @Transactional
+    public void deleteByUserId(String userId) {
+
+        userEventRepository.deleteByUserId(userId);
     }
 
     public void bookAnEvent(UserEvent userEvent) {
+        Events event = eventsRepository.findById(userEvent.getEventId()).get();
+
+        if(event.getSeatsLeft()==0){
+            throw new ObjectValidationException("'seatsLeft' can not be negative");
+        }
```

Comment:
Check this only if user is trying book seat
```suggestion
        if(event.getSeatsLeft() == 0 && bookEvent.isBooked()) {
            throw new ObjectValidationException("No seat is available. Event is full");
        }
```

Reply:
updated